src/index.py

```diff
@@ -12,23 +12,32 @@ def router(action, grid, shuffled):
         return [move(grid, action), shuffled]
 
 
-def init():
-    print('\n\n' + welcome(), end='\n\n')
-    grid = build_grid()
-    shuffled = False
+def play_one_turn(grid, shuffled):
+    print(show_grid(grid))
+    print(show_moves(movable_tiles(grid)))
+    action = input(ask_move(shuffled))
+
+    try:
+        return router(action, grid, shuffled)
+    except ValueError:
+        print('=> ' + show_action_not_valid(action))
+    except MoveException as error:
+        print('=> ' + str(error))
 
-    while True:
-        print(show_grid(grid))
-        print(show_moves(movable_tiles(grid)))
-        action = input(ask_move(shuffled))
 
-        try:
-            grid, shuffled = router(action, grid, shuffled)
-        except ValueError:
-            print('=> ' + show_action_not_valid(action))
-        except MoveException as error:
-            print('=> ' + str(error))
+def play(grid_arg, started_grid):
+    grid, shuffled = play_one_turn(grid_arg, False)
 
+    while not do_i_win(grid, started_grid):
+        grid, shuffled = play_one_turn(grid, shuffled)
+
+
+def init():
+    print('\n\n%s\n\n' % welcome())
+    grid = build_grid()
+    started_grid = grid.copy()
+    play(grid, started_grid)
+    print('\n\n%s' % victory())
```

Comment:
If the player loose or quit the game, will you have this message displayed?

Reply:
There is no way to loose the game at the moment. Moreover when the user want to quit the game, the keyboard interrupt exception is catched and a goodbye message is printed instead of the victory message.